### ISSUE TITLE
`quarto run` fails because of  `--cached-only`  and dependency version unfound

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#11219]
@cscheid This happens only with our bundled version where we activate `--cached-only`. This is why our test is passing, especially 
https://github.com/quarto-dev/quarto-cli/blob/464c18c3a4a02df537e1765883eeaf1cafa30f6d/tests/smoke/run/stdlib-run-version.test.ts#L1-L24

Trying to run the file using last prerelease I get 
````
❯ C:\Users\chris\AppData\Local\qvm\versions\v1.6.30\bin\quarto run .\tests\docs\run\test-stdlib.ts
error: JSR package version manifest for '@std/io@0.224.9' failed to load: Specifier not found in cache: "https://jsr.io/@std/io/0.224.9_meta.json", --cached-only is specified.
    at file:///C:/Users/chris/Documents/DEV_R/quarto-cli/tests/docs/run/test-stdlib.ts:1:27
````

My understanding is the following

* We are using bundling versions and it seems we do bundle `0.224.8`
  https://github.com/quarto-dev/quarto-cli/blob/464c18c3a4a02df537e1765883eeaf1cafa30f6d/src/dev_import_map.json#L19

* However the spec for `stdlib/io` used for loading the dep in a `quarto run` script is 
  https://github.com/quarto-dev/quarto-cli/blob/464c18c3a4a02df537e1765883eeaf1cafa30f6d/src/resources/deno_std/run_import_map.json#L30
  Using the `^` means that `0.224.9` is valid and it seems Deno wants do use it, but can't find it in the cache as expected.

Maybe I don't get it right, but it seems that Deno may resolve the specifier, unrelated to the cache, instead of looking for a version in cache that respect `^0.224.0` 🤔 

I did the test with `deno run`, using deno binary bundled in quarto and using this file
````ts
import { format } from "jsr:@std/io@0.224.8"
````
and in a clean cache (I deleted the all Deno cache directory)
* First run using cached only fails
  ````
  ❯ C:\Users\chris\scoop\apps\quarto-prerelease\current\bin\tools\x86_64\deno.exe run --cached-only test.ts
  error: JSR package version manifest for '@std/io@0.224.8' failed to load: Specifier not found in cache: "https://jsr.io/@std/io/0.224.8_meta.json", --cached-only is specified.
      at file:///C:/Users/chris/Documents/DEV_OTHER/00-TESTS/test-quarto/test.ts:1:24
  ````
* Second run without `--cached-only` download the dependency 
* For third run I changed to `jsr:@std/io@^0.224.0` in the file
  ````ts
  import { format } from "jsr:@std/io@^0.224.0"
  ````
  and it fails the same as with Quarto
  ````
  ❯ C:\Users\chris\scoop\apps\quarto-prerelease\current\bin\tools\x86_64\deno.exe run --cached-only test.ts
  error: Specifier not found in cache: "https://jsr.io/@std/io/0.224.9/mod.ts", --cached-only is specified.
      at file:///C:/Users/chris/Documents/DEV_OTHER/00-TESTS/test-quarto/test.ts:1:24
  ````

This is directly impacting tools like **shinylive** which needs to be updated anyway to be working with Quarto 1.6 
This is blocking https://github.com/quarto-ext/shinylive/issues/65

**Comment by cscheid** [comment]
Confirmed fixed by @cderv in the latest 1.6 prereleases.

**Comment by cscheid** [comment]
I just tested the 1.6.37 release, and this works there. Unfortunately, the first time you run a project script that imports (some) files from the standard library you might see output like this:

```
$ quarto run test.ts
Download https://jsr.io/@std/bytes/1.0.4_meta.json
Download https://jsr.io/@std/bytes/1.0.4/copy.ts
Download https://jsr.io/@std/bytes/1.0.4/concat.ts
$ quarto run test.ts
$
```

Further runs, however, will use the recently-downloaded files from the cache.
